### PR TITLE
Consider projects in confidential organizations to be confidential

### PR DESCRIPTION
**Pull request by gigxz**
And add a bunch of tests for project confidentiality

### CODE REVIEW

File: app/models/grda_warehouse/hud/project.rb [hunk 1]
```diff
@@ -206,11 +206,11 @@ class Project < Base
     end
 
     scope :confidential, -> do
-      where(confidential: true)
+      joins(:organization).where(p_t[:confidential].eq(true).or(o_t[:confidential].eq(true)))
     end
 
     scope :non_confidential, -> do
-      where(confidential: false)
+      joins(:organization).where(p_t[:confidential].eq(false).and(o_t[:confidential].eq(false)))
```
Comment (gigxz) [inline app/models/grda_warehouse/hud/project.rb:213]: Does this need to be a left join? In the database schema it doesn't look like the `organization` association is required on `project`, but it seems like we assume it to be present in many places in the code.

File: app/controllers/api/hud_filters_controller.rb [hunk 2]
```diff
@@ -30,11 +30,13 @@ def index
...
-          @data[[o_id, o_name]] ||= []
-          @data[[o_id, o_name]] << [
-            "#{name} (#{HUD.project_type_brief(type)})",
+          o_t[:confidential],
+        ).each do |id, p_name, p_confidential, type, o_name, o_id, o_confidential|
+          project_name = GrdaWarehouse::Hud::Project.confidentialize_name(current_user, p_name, p_confidential || o_confidential)
+          organization_name = o_confidential ? GrdaWarehouse::Hud::Organization.confidential_organization_name : o_name
+          @data[[o_id, organization_name]] ||= []
+          @data[[o_id, organization_name]] << [
+            "#{project_name} (#{HUD.project_type_brief(type)})",
```
Comment (gigxz) [inline app/controllers/api/hud_filters_controller.rb:39]: This is updated to also confidentialize the Organization name if applicable, which we weren't doing before. 
Screen Shot 2022-06-10 at 2 25 13 PM

File: app/models/grda_warehouse/hud/project.rb (hunk 1 above)
Comment (eanders) [reply app/models/grda_warehouse/hud/project.rb:213]: According to the spec there should never be a project without an organization.  We've definitely seen the situation where it's missing, but we should assume it's there, and it shouldn't break anything to inner join.

File: app/models/grda_warehouse/hud/project.rb [hunk 3]
```diff
@@ -573,6 +573,14 @@ def project_type_overridden_as_ph?
 
     alias_attribute :name, :ProjectName
 
+    def confidential?
+      super || organization&.confidential?
```
Comment (eanders) [inline app/models/grda_warehouse/hud/project.rb:577]: Does `super` delegate to the active record base class?  I like this if it does.

**Review by eanders — APPROVED** [review]

File: app/models/grda_warehouse/hud/project.rb (hunk 3 above)
Comment (gigxz) [reply app/models/grda_warehouse/hud/project.rb:577]: Yep it does

**Comment by gigxz** [comment]
One more update here https://github.com/greenriver/hmis-warehouse/pull/2262/commits/9b1d8ad5806dde59ca7677994e7966a5f344c48e to disable the checkbox for projects in confidential orgs @eanders. This should be good to merge if that looks ok!

Screen Shot 2022-06-13 at 8 49 16 AM
.

**Pull request merged**